src/ports/postgres/modules/svm/kernel_approximation.py_in

```diff
@@ -0,0 +1,481 @@
+                         data=self.rd_offset,
+                         data_type=data_type))
+
+    def __del__(self):
+        self.clear()
+
+    def saveAs(self, name):
+        if self.orig_data:
+            plpy.warning("Gaussian Kernel Warning: no need to save."
+                         "Original data table exists: {0}"
+                         .format(self.orig_data))
+            return
+
+        run_sql = """
+            create table {name} as
+                select
+                        {id} as id, {val} as val,
+                        'offsets' as desp
+                from {rd_offset}
+                union
+                select
+                        {id} as id, {val} as val,
+                        'weights' as desp
+                from {rd_weights}
+        """.format(name=name,
+                   id=self.rd_id,
+                   val=self.rd_val,
+                   rd_offset=self.rd_offset,
+                   rd_weights=self.rd_weights)
+        plpy.execute(run_sql)
+
+    @classmethod
+    def parse_params(cls, kernel_params='', n_features=10):
+        params_default = {
+            'in_memory': 1,
+            'gamma': 1/n_features,
+            'random_state': 1,
+            'n_components': 2*n_features}
+        params_types = {
+            'in_memory': int,
+            'gamma': float,
+            'random_state': int,
+            'n_components': int}
+        return extract_keyvalue_params(kernel_params,
+                                       params_types,
+                                       params_default)
+
+    @classmethod
+    def create(cls, schema_madlib, n_features, kernel_params):
+        params = cls.parse_params(kernel_params, n_features)
+        in_memory = params.pop('in_memory', True)
+        # according to the 1gb limit on each entry of the table
+        nelems = params['n_components']*n_features
+        if in_memory and nelems <= 1e8:
+            return GaussianKernelInMemory(schema_madlib, **params)
+        else:
+            return GaussianKernel(schema_madlib, **params)
+
+    @classmethod
+    def loadFrom(cls, schema_madlib, data, kernel_params=''):
+        rd_weights = unique_string(desp='random_weights')
+        rd_offset = unique_string(desp='random_offsets')
+        rd_val = unique_string(desp='val')
+        rd_id = unique_string(desp='id')
+        plpy.execute("""
+                drop view if exists {rd_weights};
+                create temp view {rd_weights} as
+                    select id as {rd_id}, val as {rd_val} from {data}
+                    where desp = 'weights';
+
+                drop view if exists {rd_offset};
+                create temp view {rd_offset} as
+                    select id as {rd_id}, val as {rd_val} from {data}
+                    where desp = 'offsets';
+                     """.format(**locals()))
+        params = cls.parse_params(kernel_params)
+        in_memory = params.pop('in_memory', True)
```

Comment:
I understand. I used **kwargs in the base class but in the derived class I am trying to make the interface more restrictive such that no unexpected variables got passed to base class through, for example, super().__init__(**locals()). Plus, it will be more clear to the caller this way what are expected to create the object